tests/Microsoft.Bot.Builder.ApplicationInsights.Tests/BotTelemetryClientTests.cs

```diff
@@ -3,76 +3,138 @@
+                var telemetryClient = new TelemetryClient();
 
+                var botTelemetryClient = new BotTelemetryClient(telemetryClient);
+            }
         }
-        
-        [TestMethod]
-        public void TrackAvailabilityTest()
+
+        [TestClass]
+        public class TrackTelemetryTests
         {
-            // Just invoke underlying TelemetryClient.  Not configured, just tests if it can be invoked.
-            // Class is sealed, so nothing to mock here.
-            var telemetryClient = new TelemetryClient();
-            var client = new BotTelemetryClient(telemetryClient);
+            private BotTelemetryClient _botTelemetryClient;
+            private Mock<ITelemetryChannel> _mockTelemetryChannel;
 
-            client.TrackAvailability("test", DateTimeOffset.Now, new TimeSpan(1000), "run location", true,
-                "message", new Dictionary<string, string>() { { "hello", "value" } }, new Dictionary<string, double>() { { "metric", 0.6 } });
-        }
+            [TestInitialize]
+            public void TestInitialize()
+            {
+                _mockTelemetryChannel = new Mock<ITelemetryChannel>();
 
+                var telemetryConfiguration = new TelemetryConfiguration("UNITTEST-INSTRUMENTATION-KEY", _mockTelemetryChannel.Object);
+                var telemetryClient = new TelemetryClient(telemetryConfiguration);
 
-        [TestMethod]
-        public void TrackEventTest()
-        {
-            // Just invoke underlying TelemetryClient.  Not configured, just tests if it can be invoked.
-            // Class is sealed, so nothing to mock here.
-            var telemetryClient = new TelemetryClient();
-            var client = new BotTelemetryClient(telemetryClient);
+                _botTelemetryClient = new BotTelemetryClient(telemetryClient);
+            }
 
-            client.TrackEvent("test", new Dictionary<string, string>() { { "hello", "value" } }, new Dictionary<string, double>() { { "metric", 0.6 } });
-        }
+            [TestMethod]
+            public void TrackAvailabilityTest()
+            {
+                _botTelemetryClient.TrackAvailability("test", DateTimeOffset.Now, new TimeSpan(1000), "run location", true,
+                    "message", new Dictionary<string, string>() { { "hello", "value" } }, new Dictionary<string, double>() { { "metric", 0.6 } });
 
-        [TestMethod]
-        public void TrackDependencyTest()
-        {
-            // Just invoke underlying TelemetryClient.  Not configured, just tests if it can be invoked.
-            // Class is sealed, so nothing to mock here.
-            var telemetryClient = new TelemetryClient();
-            var client = new BotTelemetryClient(telemetryClient);
-            client.TrackDependency("test", "target", "dependencyname", "data", DateTimeOffset.Now, new TimeSpan(10000), "result", false );
-        }
+                _mockTelemetryChannel.Verify(tc => tc.Send(It.Is<AvailabilityTelemetry>(t =>
```

Comment:
Just a thought, but is there a way to work the Assert.AreEquals calls into here - because then the failure of the individual assert indicates what is potentially wrong. The && groups everything together - certainly the test fails but you get less of an indicator as to why.